02-peer-protocol.md

```diff
@@ -2529,6 +2530,9 @@ A receiving node:
     - otherwise:
       - MUST send `tx_abort` to let the sending node know that they can forget
         this funding transaction.
+  - if `next_funding_txid` is not set, and `next_commitment_number` is zero:
+    - MUST immediately fail the channel and broadcast any relevant latest commitment
+      transaction.
```

Comment:
LND will respond to a normal establishment msg with an error so there is no need for the other implementation to immediately force close, but I think having this code part in place does also not violate the spec am I correct ?

Reply:
Ok understand so you mean the first message from the peer can immediately be a error instead of a `establishment-msg` ? In that case LND would not force-close according to the code right now.

Reply:
> Alice then receives from her peer a channel_reestablish with next_commitment_number > 0. Will Alice automatically send back an error message to request her peer to force-close the channel?

yes 

> If she does and the peer is lnd, will it force-close when receiving the error message?

No because we encountered some errors with CLN nodes see for more background
https://github.com/ElementsProject/lightning/issues/4212, that's why we currently handle the errors in a soft way.

Can somebody form CLN look whether CLN still sends an error if there are gossip sync issue ?

Reply:
Triggering a force-close with a channel-reestablishment (setting next_commitment_number = 0) msg seems more robust rather then always force closing when an error is received wdyt ?